ami_deprecation_tool/api.py

```diff
@@ -114,7 +122,15 @@ def _apply_deprecation_policy(
 
     image_actions = ActionImages()
 
-    for image in sorted(list(region_images.keys()), reverse=True):
+    sorted_images = sorted(list(region_images.keys()), reverse=True)
+
+    if policy.expiration_months:
+        cutoff = dt.datetime.now() - relativedelta(months=policy.expiration_months)
+        sorted_images = list(filter(lambda i: region_images[i][0].creation_date > cutoff, sorted_images)) or [
+            sorted_images[0]
```

Comment:
should the `or` be `sorted_images[:policy.keep]`? this seems like it wouldn't ever match the keep policy, but i may be missing something